qemu/tests/sr_iov_sanity.py

```diff
@@ -104,30 +112,34 @@ def run(test, params, env):
     ethname_dict = []
     ips = {}
 
-    msg = "Configure all VFs in host."
-    error.context(msg, logging.info)
-    for pci_id in vf_pci_id:
-        cmd = "ls /sys/bus/pci/devices/%s/net/" % pci_id
-        ethname = utils.system_output(cmd).strip()
-        ethname_dict.append(ethname)
-        network_script = os.path.join("/etc/sysconfig/network-scripts",
-                                      "ifcfg-%s" % ethname)
-        if not os.path.exists(network_script):
-            error.context("Create %s file." % network_script, logging.info)
-            txt = "DEVICE=%s\nONBOOT=yes\nBOOTPROTO=dhcp\n" % ethname
-            file(network_script, "w").write(txt)
-
-    msg = "Check whether VFs could get ip in host."
-    error.context(msg, logging.info)
-    for ethname in ethname_dict:
-        ifup_down_interface(ethname)
-        _ip = check_network_interface_ip(ethname)
-        if not _ip:
-            msg = "Interface '%s' could not get IP." % ethname
-            logging.error(msg)
-        else:
-            ips[ethname] = _ip
-            logging.info("Interface '%s' get IP '%s'", ethname, _ip)
+    # Not all test environments would have a dhcp server to serve IP for
+    # all mac addresses. So configure_on_host param has been
+    # introduced to choose whether configure VFs on host or not
+    if (configure_on_host):
+        msg = "Configure all VFs in host."
+        error.context(msg, logging.info)
+        for pci_id in vf_pci_id:
+            cmd = "ls /sys/bus/pci/devices/%s/net/" % pci_id['vf_id']
+            ethname = utils.system_output(cmd).strip()
+            ethname_dict.append(ethname)
+            network_script = os.path.join("/etc/sysconfig/network-scripts",
+                                          "ifcfg-%s" % ethname)
+            if not os.path.exists(network_script):
```

Comment:
this might not work for other distributions like ubuntu, you can consider using similar function in this PR,https://github.com/avocado-framework/avocado-vt/pull/825/